pkg/jobs/job_info_storage.go

```diff
@@ -0,0 +1,250 @@
+// Copyright 2023 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package jobs
+
+import (
+	"bytes"
+	"context"
+
+	"github.com/cockroachdb/cockroach/pkg/sql/isql"
+	"github.com/cockroachdb/cockroach/pkg/sql/sem/tree"
+	"github.com/cockroachdb/cockroach/pkg/sql/sessiondata"
+	"github.com/cockroachdb/cockroach/pkg/sql/sqlliveness"
+	"github.com/cockroachdb/cockroach/pkg/util/log"
+	"github.com/cockroachdb/cockroach/pkg/util/tracing"
+	"github.com/cockroachdb/errors"
+)
+
+// InfoStorage can be used to read and write rows to system.jobs_info table. All
+// operations are scoped under the txn and are are executed on behalf of Job j.
+type InfoStorage struct {
+	j   *Job
+	txn isql.Txn
+}
+
+// InfoStorageWithTxn returns a new InfoStorage with the passed in job and txn.
+func (j *Job) InfoStorageWithTxn(txn isql.Txn) InfoStorage {
+	return InfoStorage{j: j, txn: txn}
+}
+
+// InfoStorageNoTxn returns a new InfoStorage with the passed in job but no txn.
+func (j *Job) InfoStorageNoTxn() InfoStorage {
```

Comment:
Is this just for tests?

Reply:
If we really want this, I think I'd go separate wrapper instead of the recursive methods

Reply:
Deleted the NoTxn variant, if and when we need it we can go the wrapper way you suggested to me.